python/pyspark/sql/dataframe.py

```diff
@@ -282,8 +284,18 @@ def show(self, n=20, truncate=True):
         |  2|Alice|
         |  5|  Bob|
         +---+-----+
-        """
-        print(self._jdf.showString(n, truncate))
+        >>> df.show(truncate=3)
+        +---+----+
+        |age|name|
+        +---+----+
+        |  2| Ali|
+        |  5| Bob|
+        +---+----+
+        """
+        if truncate == 1:
```

Comment:
Yes, I did it because, in python 1 and True are indistinguishable.
```python
Type "help", "copyright", "credits" or "license" for more information.
>>> a=True
>>> isinstance(a, bool)
True
>>> isinstance(a, int)
True
>>> a == 1
True
>>> a == True
True
>>> 

```

In both python 2 and 3. 
But,
```python
>>> a=1
>>> isinstance(a, bool)
False
```
Might help here. I can go ahead and change it.